packages/fuels-core/src/abi_encoder.rs

```diff
@@ -141,6 +215,23 @@ impl ABIEncoder {
                 CodecError::InvalidData(msg)
             })
     }
+
+    fn encode_vector(data: &[Token]) -> Result<Vec<Data>, CodecError> {
+        let encoded_data = Self::encode_tokens(data)?;
+        let cap = data.len() as u64;
+        let len = data.len() as u64;
+
+        // A vector is expected to be encoded as 3 WORDs -- a ptr, a cap and a
+        // len. This means that we must place the encoded vector elements
+        // somewhere else. Hence the use of Data::Dynamic which will, when
+        // resolved, leave behind in its place only a pointer to the actual
+        // data.
+        Ok(vec![
+            Data::Dynamic(encoded_data),
+            Self::encode_u64(cap),
+            Self::encode_u64(len),
+        ])
+    }
```

Comment:
![image](https://user-images.githubusercontent.com/3484025/193430276-409d3ade-7e99-47d8-a218-0b5525cadd80.png)

Very nice abstraction you got going on here!